src/cronjobs/dex/swaps/swaps-tasks.ts

```diff
@@ -109,3 +113,55 @@ async function prepareSdexSwaps(chain: SdexChain, chainId: number) {
     childLogger.error(error, 'Error while retrieving Swaps V2 for Sdex chain');
   }
 }
+
+async function prepareLegacySwaps(chain: LegacyChain, chainId: number) {
+  try {
+    const blockNumber = await chain.queryBlockNumber();
+    const items = await chain.querySwaps(blockNumber);
```

Comment:
limits which blocks to load, otherwise you will have only last 1000

```suggestion
    const lastSwap = await swapRepositoryV2.loadLastSwap(chainId); // Load the last swap from the repository
    const startBlock = lastSwap ? lastSwap.block - 100 : chain.startBlock; // Determine the start block for querying
    const items = await chain.querySwaps(startBlock, blockNumber);
```

Reply:
also, add `chainId` prop for `swapRepositoryV2.loadLastSwap(chainId)` on the `prepareSdexSwaps` function too